solidity/contracts/integrator/TBTCDepositorProxy.sol

```diff
@@ -0,0 +1,309 @@
+    /// @param fundingTx Bitcoin funding transaction data, see `IBridgeTypes.BitcoinTxInfo`.
+    /// @param reveal Deposit reveal data, see `IBridgeTypes.DepositRevealInfo` struct.
+    /// @param extraData 32-byte deposit extra data.
+    /// @return depositKey Deposit key computed as
+    ///         `keccak256(fundingTxHash | reveal.fundingOutputIndex)`. This
+    ///         key can be used to refer to the deposit in the Bridge and
+    ///         TBTCVault contracts.
+    /// @dev Requirements:
+    ///      - The revealed vault address must match the TBTCVault address,
+    ///      - All requirements from {Bridge#revealDepositWithExtraData}
+    ///        function must be met.
+    // slither-disable-next-line dead-code
+    function _initializeDeposit(
+        IBridgeTypes.BitcoinTxInfo calldata fundingTx,
+        IBridgeTypes.DepositRevealInfo calldata reveal,
+        bytes32 extraData
+    ) internal returns (uint256) {
+        require(reveal.vault == address(tbtcVault), "Vault address mismatch");
+
+        uint256 depositKey = _calculateDepositKey(
+            _calculateBitcoinTxHash(fundingTx),
+            reveal.fundingOutputIndex
+        );
+
+        pendingDeposits[depositKey] = true;
+
+        emit DepositInitialized(
+            depositKey,
+            /* solhint-disable-next-line not-rely-on-time */
+            uint32(block.timestamp)
+        );
+
+        // The Bridge does not allow to reveal the same deposit twice and
+        // revealed deposits stay there forever. The transaction will revert
+        // if the deposit has already been revealed so, there is no need to do
+        // an explicit check here.
+        bridge.revealDepositWithExtraData(fundingTx, reveal, extraData);
+
+        return depositKey;
+    }
+
+    /// @notice Finalizes a deposit by calculating the amount of TBTC minted
+    ///         for the deposit and calling the `onDepositFinalized` callback
+    ///         function.
+    /// @param depositKey Deposit key identifying the deposit.
+    /// @return tbtcAmount Approximate amount of TBTC minted for the deposit.
+    /// @return extraData 32-byte deposit extra data.
+    /// @dev Requirements:
+    ///      - The deposit must be initialized but not finalized
+    ///        (in the context of this contract) yet.
+    ///      - The deposit must be finalized on the Bridge side. That means the
+    ///        deposit must be either swept or optimistically minted.
+    /// @dev IMPORTANT NOTE: The tbtcAmount returned by this function is an
+    ///      approximation. See documentation of the `calculateTbtcAmount`
+    ///      responsible for calculating this value for more details.
+    // slither-disable-next-line dead-code
+    function _finalizeDeposit(uint256 depositKey)
+        internal
+        returns (uint256 tbtcAmount, bytes32 extraData)
+    {
+        require(pendingDeposits[depositKey], "Deposit not initialized");
```

Comment:
Do we need `pendingDeposits` mapping? I think it should be enough to check whether [the `deposit.depositor` value](https://github.com/keep-network/tbtc-v2/blob/609b6fedc9444688092f310c38c0e03758bfdbfb/solidity/contracts/bridge/Deposit.sol#L102) is non-zero or set to `address(this)`. We retrieve `IBridgeTypes.DepositRequest` anyway. Also, the implementation of the `TBTCDepositorProxy` is already tightly coupled with the implementation of `Bridge` and `TBTCVault`. We could save some gas especially since those functions will be called by a relayer bot and not by the users.

EDIT: No, the current version is correct. We do not want `_finalizeDeposit` to be called twice for the same deposit.

Reply:
> EDIT: No, the current version is correct. We do not want _finalizeDeposit to be called twice for the same deposit.

Exactly. Moreover, we don't want to push this responsibility to the child contract.